commit/src/domain.rs

```diff
@@ -9,58 +9,111 @@ use p3_matrix::dense::RowMajorMatrix;
+///
+/// Each `PolynomialSpace` should be part of a generic family of similar spaces for some
+/// collection of sizes (usually powers of two). Any space other than at the smallest size
+/// should be decomposable into a disjoint collection of smaller spaces. Additionally, the
+/// set of all `PolynomialSpace` of a given size should form a disjoint partition of some
+/// subset of `F^n` which supports a group structure.
+///
+/// The canonical example of a `PolynomialSpace` is a coset `gH` of
+/// a two-adic subgroup `H` of the multiplicative group `F*`. This satisfies the properties
+/// above as cosets partition the group and decompose as `gH = g(H^2) u gh(H^2)` for `h` a
+/// generator of `H`.
 pub trait PolynomialSpace: Copy {
+    /// The base field `F`.
     type Val: Field;
 
+    /// The number of elements of the space.
     fn size(&self) -> usize;
 
+    /// The first point in the space.
     fn first_point(&self) -> Self::Val;
 
-    // This is only defined for cosets.
+    /// An algebraic function which takes the i'th element of the space and returns
+    /// the (i+1)'th evaluated on the given point.
+    ///
+    /// When `PolynomialSpace` corresponds to a coset, `gH` this
+    /// function is multiplication by `h` for a chosen generator `h` of `H`.
+    ///
+    /// This function may not exist for other classes of `PolynomialSpace` in which
+    /// case this will return `None`.
     fn next_point<Ext: ExtensionField<Self::Val>>(&self, x: Ext) -> Option<Ext>;
 
-    // There are many choices for this, but we must pick a canonical one
-    // for both prover/verifier determinism and LDE caching.
+    /// Return another `PolynomialSpace` with size `min_size` disjoint from this space.
+    ///
+    /// This fixes a canonical choice for prover/verifier determinism and LDE caching.
     fn create_disjoint_domain(&self, min_size: usize) -> Self;
 
-    /// Split this domain into `num_chunks` even chunks.
-    /// `num_chunks` is assumed to be a power of two.
+    /// Split the `PolynomialSpace` into `num_chunks` smaller `PolynomialSpaces` of equal size.
+    ///
+    /// `num_chunks` must divide `self.size()` (which usually forces it to be a power of 2.)
     fn split_domains(&self, num_chunks: usize) -> Vec<Self>;
-    // Split the evals into chunks of evals, corresponding to each domain
-    // from `split_domains`.
+
+    /// Split a set of polynomial evaluations over this `PolynomialSpace` into a vector
+    /// of polynomial evaluations over each `PolynomialSpace` generated from `split_domains`.
     fn split_evals(
         &self,
         num_chunks: usize,
         evals: RowMajorMatrix<Self::Val>,
     ) -> Vec<RowMajorMatrix<Self::Val>>;
 
+    /// Compute the zerofier of the space, evaluated at the given point.
     fn zp_at_point<Ext: ExtensionField<Self::Val>>(&self, point: Ext) -> Ext;
 
-    // Unnormalized
+    /// Compute several lagrange selectors at a given point.
```

Comment:
"several" is a bit vague. Where do the selectors come from?
Same goes for the next function.